runtime/domain.c

```diff
@@ -1827,16 +1827,19 @@ static void handover_ephemerons(caml_domain_state* domain_state)
   CAMLassert (domain_state->ephe_info->todo == 0);
 }
 
-static void handover_finalisers(caml_domain_state* domain_state)
+static void orphan_finalisers(caml_domain_state* domain_state)
 {
   struct caml_final_info* f = domain_state->final_info;
 
   if (f->todo_head != NULL || f->first.size != 0 || f->last.size != 0) {
     /* have some final structures */
     if (caml_gc_phase != Phase_sweep_and_mark_main) {
-      /* Force a major GC cycle to simplify constraints for
-       * handing over finalisers. */
+      /* Force a major GC cycle to simplify constraints for orphaning
+         finalisers. See note attached to the declaration of
```

Comment:
The comment attached to `num_domains_orphaning_finalisers` explains the motivation behind it. I guess the word `constraint` is a problem here since choosing to only orphan and adopt finalisers in `Phase_sweep_and_mark_main` is a strategy to simplify the code, and not a "constraint".